source/agora/node/Ledger.d

```diff
@@ -585,8 +585,12 @@ public class Ledger
         @safe
     {
         data = ConsensusData.init;
-        data.timestamp = max(clock.networkTime(), this.last_block.header.timestamp + 1);
-        log.trace("Going to nominate current timestamp [{}] or newer", clock.networkTime());
+        const ulong genesisTimestamp = this.params.GenesisTimestamp;
+        const ulong now = clock.networkTime;
+        const ulong genesis_offset = now - genesisTimestamp;
+        log.trace("genesisTimestamp:{} now:{} genesis_offset:{} last block: {}", genesisTimestamp, now, genesis_offset, this.last_block.header.time_offset);
+        data.time_offset = max(genesis_offset, this.last_block.header.time_offset + 1);
+        log.trace("Going to nominate current time_offset [{}] or newer", data.time_offset);
```

Comment:
Let's not have two consecutive log messages without branching

Reply:
Forgot to remove. Updated.